src/ucp/core/ucp_request.inl

```diff
@@ -91,28 +91,98 @@ void ucp_request_recv_generic_dt_finish(ucp_request_t *req)
     }
 }
 
-static UCS_F_ALWAYS_INLINE ucs_status_t
-ucp_request_send_buffer_reg(ucp_request_t *req, ucp_lane_index_t lane)
+static UCS_F_ALWAYS_INLINE
+void ucp_iov_buffer_memh_dereg(uct_md_h uct_md, uct_mem_h *memh,
```

Comment:
let's put those 3 functions in C file, this is not fast path, would reduce the size of the send function code and instruction cache misses